Assets/General/Audio/FMODScripts/IntervalAmbienceEvents.cs

```diff
@@ -0,0 +1,63 @@
+/******************************************************************************
+// File Name:       IntervalAmbienceEvents.cs
+// Author:          Mark Hanson
+// Creation Date:   11/14/2024
+//
+// Description:     Handler of random ambient sound effects at random times
+******************************************************************************/
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using FMODUnity;
+using Random = UnityEngine.Random;
+/// <summary>
+/// Handler of random ambient sound effects at random times
+/// </summary>
+public class IntervalAmbienceEvents : MonoBehaviour
+{
+  /// <summary>
+  /// Struct to hold min and max time for random range
+  /// </summary>
+    public struct FModEventReference
+    {
+        public int MinTimeBetweenEvents;
+        public int MaxTimeBetweenEvents;
+    }
+
+    public FModEventReference FmodEvents { get; set; }
+   [SerializeField] private EventReference[] _ambienceArray;
+
+   private Coroutine _ambienceCoroutine;
+   /// <summary>
+   /// Start coroutine 
+   /// </summary>
+   private void Start()
+   {
+       if (_ambienceCoroutine == null)
+       {
+           StartCoroutine(PlayIntervalAudio());  
+       }
+   }
+/// <summary>
+/// The Coroutine to loop through the array with different timers per sound call
+/// </summary>
+/// <returns></returns>
+   private IEnumerator PlayIntervalAudio()
+    {
+        while (true)
+        {
+            float Timer = 0.0f;
+            for (int i = 0; i < _ambienceArray.Length; i++)
```

Comment:
The way this is setup confuses me. To translate this whole thing to pseudocode: In an infinite coroutine loop we are iterating through every value in the ambience array. Then we are checking if that value is more than a random value between the min and max time between events. We are then playing the sound effect at the location of this script. This leaves me with a few questions. Wouldn't it just reset the timer to 0 every frame so it would never actually get to the point of the timer being more than the range? Why are we creating a new random number every frame for each value in the array?